grails-datastore-gorm/src/main/groovy/org/grails/datastore/gorm/finders/MethodExpression.java

```diff
@@ -258,7 +264,12 @@ public InList(String propertyName) {
 
         @Override
         public Query.Criterion createCriterion() {
-            return Restrictions.in(propertyName, (Collection<?>) arguments[0]);
+            Collection<?> argument = (Collection<?>) arguments[0];
+            if (argument == null || argument.isEmpty()) {
+                return Restrictions.and(Restrictions.isNull(propertyName), Restrictions.isNotNull(propertyName));
```

Comment:
I don't understand why `Restrictions.and(Restrictions.isNull(propertyName), Restrictions.isNotNull(propertyName))` would make sense.  Will that always effectively be `false`?  If so, surely there is a better way than adding to the criteria these 2 conditions.

Reply:
Yes, always false. But I don't know how to do it better. 
https://github.com/grails/grails-data-mapping/issues/1212#issuecomment-455879135

Reply:
Dear @jeffbrown , I would be very grateful if you could tell me how to make it easier. I have not found easy solution